### PR TITLE
Make the compile work on a Mac

### PR DESCRIPTION
**Pull request by ballanty**
This change:

- sed as used required the addition of a small change to have it work on the Mac. The Makefile was altered to detect the platform and conditionally use sed in different ways in one case.
- it looks like "packets" was once a buffer size measure and it was changed to bytes at some point.  Fix what appears to be a leftover reference to packets.
- parsing of quantities ("2days" for example) seemed to differ by platform (Mac (didn't work) vs Linux (did)).  Replaced parsing approach with one using regular expressions to make it more robust to platform differences between the way std::..stream operator>> parsed.

### CODE REVIEW

File: src/quantity.cpp [hunk 1]
```diff
@@ -20,8 +21,19 @@ Quantity::Quantity(const MeasurementUnit &converter_) :
...
+    std::smatch match;
+    std::regex_search(str, match, reg);
+
+    if ( !match.empty() ) {
+        this->value = std::stod(match[1].str() + match[2].str());
+        this->unit = match[3];
+    } else {
+        this->value = 0.0;
+        this->unit = "";
+    }
```
Comment (lorenzosaino) [inline src/quantity.cpp]: Minor nit: this block uses spaces for indentation, while the rest of the code uses tabs. I'd suggest we use tabs here too for consistency

File: src/quantity.cpp [hunk 2]
```diff
@@ -20,8 +21,19 @@ Quantity::Quantity(const MeasurementUnit &converter_) :
...
 void Quantity::fromString(const std::string &str) {
-	std::istringstream ss(str);
-	ss>>this->value>>this->unit;
+
+    std::string regexpstr = "([0-9]+)(\\.[0-9]*)? *([^ \t]*)";
+    std::regex reg(regexpstr);
+    std::smatch match;
+    std::regex_search(str, match, reg);
+
+    if ( !match.empty() ) {
```
Comment (lorenzosaino) [inline src/quantity.cpp]: Minor nit: in the rest of the code, there are not spaces between parentheses and conditions (see for example at L38 below). I would use the same convention for consistency

File: test/test.cpp [hunk 3]
```diff
@@ -520,11 +522,11 @@ void testEvent() {
 
 void testQuantity() {
 	cout<<"Quantity test: ";
-	Quantity t1(Units::Time);
+	Quantity t1("2days",Units::Time);
 	Quantity t2(1, "h", Units::Time);
 	Quantity t3("60min", Units::Time);
 	Quantity t4("3601 sec", Units::Time);
-	t1.fromString("2days");
+	//t1.fromString("2days");
```
Comment (lorenzosaino) [inline test/test.cpp]: Is commenting this out intended? Without this we are no longer testing anywhere the `fromString` method

File: Makefile [hunk 4]
```diff
@@ -159,8 +167,8 @@ run_test    = printf "Executing:   $1:  "; \
              if [ $$OUT -eq 0 ]; then \
                  echo "OK"; \
              else \
-                 rm $1; \
-                 exit 1; \
+                  rm $1; \
+                  exit 1; \
```
Comment (lorenzosaino) [inline Makefile]: Minor nit: not sure if intended but by adding space in front of these two lines makes them no longer aligned with the `echo "OK"; \ ` instruction above. Was this change intended?

File: test/test.cpp [hunk 5]
```diff
@@ -302,6 +302,8 @@ void testEdge() {
 	cout<<"Edge test: ";
 
 	Edge e;
+
+    e.setBufferSize(Quantity("1kB", Units::Data));
```
Comment (lorenzosaino) [inline test/test.cpp]: Minor nit like my comment above about using tabs instead of spaces

**Review by lorenzosaino — CHANGES_REQUESTED** [review]
Thanks for contributing this feature. I made a few comments about minor nits, but the core change looks good.

**Comment by ballanty** [comment]
Good catches all.  I'm sorry I made you go through the detailed review.  I'll remember the project conventions going forward!

**Review by lorenzosaino — APPROVED** [review]
Looks great, thank you!
